samtranslator/model/eventsources/pull.py

```diff
@@ -181,11 +203,69 @@ class MSK(PullEventSource):
     def get_policy_arn(self):
         return ArnGenerator.generate_aws_managed_policy_arn("service-role/AWSLambdaMSKExecutionRole")
 
+    def get_policy_statements(self):
+        return None
+
 
 class MQ(PullEventSource):
     """MQ event source."""
 
     resource_type = "MQ"
 
     def get_policy_arn(self):
-        return ArnGenerator.generate_aws_managed_policy_arn("service-role/AWSLambdaAMQExecutionRole")
+        return None
+
+    def get_policy_statements(self):
+        if not self.SourceAccessConfigurations:
+            raise InvalidEventException(
```

Comment:
Can we check how `InvalidEventException` been handled? I remember we are using something like `InvalidDocumentException` for validations, and wondering if `InvalidEventException` will result in `4xx` error.

Reply:
File "/Users/qingchm/serverless-application-model/bin/../samtranslator/parser/parser.py", line 12, in parse
    self._validate(sam_template, parameter_values)
  File "/Users/qingchm/serverless-application-model/bin/../samtranslator/parser/parser.py", line 30, in _validate
    raise InvalidDocumentException([InvalidTemplateException("'Resources' section is required")])
samtranslator.model.exceptions.InvalidDocumentException: [InvalidTemplateException("'Resources' section is required")]

Reply:
In both `pull.py` and `push.py`, `InvalidEventException` was used instead of `InvalidDocumentException`. 
I think it's ambiguous, but `InvalidEventException` might refer to an exception in an event source of a function

Reply:
So looking at the code, `InvalidEventException` have been caught in here and `InvalidResourceException` have been thrown; https://github.com/aws/serverless-application-model/blob/develop/samtranslator/model/sam_resources.py#L193

And `InvalidResourceException` have been caught here, and errors have been appended; https://github.com/aws/serverless-application-model/blob/develop/samtranslator/translator/translator.py#L145

Looking at these, it should work as expected. And I guess this `InvalidEventException` is created specific to function event sources.

Reply:
Agreed. And I think the docstring of `InvalidEventException` should be better written.